### DISCUSSION TITLE
How to convert the private key

### DISCUSSION THREAD

**Discussion by Aaron-018** [trustwallet/wallet-core#3817] [Q&A]
Is there a JavaScript method available to convert an EVM format private key to a Solana format private key?

**Comment by satoshiotomakan** [comment]
Hi @Aaron-018, it depends on your purpose.
Basically, you can create a `PrivateKey` from the secp56k1 (used in EVM) 32-length byte array private key data, and just use it for Solana transaction signing or generating address (ed25519).
Private key is mostly a scalar, and most (if not all) of the secp256k1 scalars belong to the ed25519 elliptic curve
